### DISCUSSION TITLE
Curso Fundamentos do SQL SERVER - SQL SERVER em DOCKER

### DISCUSSION THREAD

**Discussion by darkrepulsor** [balta-io/forum#358] [Tira Dúvidas] [answered]
Boa noite, pessoal!!!

Estou tentando seguir o curso como o balta mostra, porém não estou conseguindo conectar no banco de dados que está no docker, a mensagem retornada no Azure Data Studio é a seguinte:
Login failed for user 'sa'.

Nos logs do container:
2023-07-07 22:55:27.58 Logon       Error: 18456, Severity: 14, State: 7.
2023-07-07 22:55:27.58 Logon       Login failed for user 'sa'. Reason: An error occurred while evaluating the password. [CLIENT: 172.17.0.1]

A senha para acessar o Banco é a seguinte:
1A<3T&e2023

OBS.: Já tenho uma versão do SQL SERVER 19 para desenvolvedor, instalado na máquina e costumo utilizar o MSSQL para manipular os dados. 

![image](https://github.com/balta-io/forum/assets/25958548/c45bcf8d-fe6a-4473-aa20-5f57bd850fd9)

![image](https://github.com/balta-io/forum/assets/25958548/2b647588-f17d-413a-bae7-1cfe456ea545)

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @darkrepulsor como vai?
Esta mensagem diz que seu usuário ou senha são inválidos, provavelmente algum caractere, espaço, ou algo do tipo.

Lembrando que se você já tem o SQL Server local, não precisa do Docker..Pode se conectar a ele mesmo.

**Reply by darkrepulsor** [reply]
Balta, encontrei o erro, seguindo esse link!

https://github.com/Microsoft/mssql-docker/issues/301

Estava passando entre aspas as credenciais e por algum motivo, não conectava com o SA.
Seguindo o que está no blog [SQL Server Docker - Balta](https://balta.io/blog/sql-server-docker):
docker run --name sqlserver -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWAORD=#1AT3e2023" -p 1433:1433 -d mcr.microsoft.com/mssql/server

Quando removi as aspas, consegui conectar:
docker run --name sqlserver -e ACCEPT_EULA=Y -e MSSQL_SA_PASSWORD=#1AT3e2023 -p 1433:1433 mcr.microsoft.com/mssql/server

![image](https://github.com/balta-io/forum/assets/25958548/8ea9f476-9902-44b6-b1a4-40d212d0ff33)

**Reply by andrebaltieri** [reply]
Showww!!

**Reply by oimarcella** [reply]
Boa, funcionou para mim também

**Comment by luizgparaujo** [comment]
Eu também tive o mesmo problema durante a conexão e conto com o mesmo cenário do @darkrepulsor, o que fiz foi parar a execução do serviço MSSQL$SQLEXPRESS na minha máquina e na sequência refiz todos os passos do artigo https://balta.io/blog/sql-server-docker e funcionou.

![image](https://github.com/balta-io/forum/assets/52684053/e55325d7-b177-49ed-bd54-01d405b85441)


![image](https://github.com/balta-io/forum/assets/52684053/b2831249-29b4-4c2d-b1fd-2ec779fe5e29)

**Comment by ricardoo-cavalheiro** [comment]
Tive o mesmo problema. No meu caso, não usei a linha de comando para inicar o container, usei a GUI do Docker. O problema foi na hora de definir o nome das variáveis de ambiente. Apertei sem querer CTRL + V duas vezes.

![image](https://github.com/balta-io/forum/assets/81830062/dbcbfaf8-c3eb-42d1-8835-50fde9d6d30b)

**Comment by lgc2** [comment]
Fiz 80% do módulo com o DBeaver e foi maravilha, zero erros. No fim, tive que mudar para a IDE do Azure para poder fazer restore com o arquivo 'balta.bacpac', e agora não consigo me conectar de jeito nenhum. Como obtenho ajuda?
Com exatamente os mesmo dados, consigo conectar pelo DBeaver normalmente.

![image](https://github.com/balta-io/forum/assets/87617474/99a5abf1-5901-493a-b44a-fa17b2b52192)

**Reply by lgc2** [reply]
[Resolvido]
foi só trocar `localhost` por `127.0.0.1`
